models/player.go

```diff
@@ -312,6 +312,56 @@ func GetPlayerDetails(db DB, encryptionKey []byte, gameID, publicID string) (map
 	return result, nil
 }
 
+// GetPlayersToEncrypt get players that have plain text name
+func GetPlayersToEncrypt(db DB, encryptionKey []byte, amount int) ([]*Player, error) {
+	query := `SELECT p.*
+	FROM players p
+		LEFT JOIN encrypted_players ep ON p.id = ep.player_id
+	WHERE ep.player_id IS NULL
+	LIMIT $1`
+
+	var players []*Player
+	_, err := db.Select(&players, query, amount)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, player := range players {
+		decryptedName, err := util.DecryptData(player.Name, encryptionKey)
+		if err != nil {
+			continue
+		}
+
+		player.Name = decryptedName
+	}
+
+	return players, nil
+}
+
+// ApplySecurityChanges encrypt and update player
+func ApplySecurityChanges(db DB, encryptionKey []byte, players []*Player) error {
+	for _, player := range players {
+		encryptedName, err := util.EncryptData(player.Name, encryptionKey)
+		if err != nil {
+			return err
+		}
+
+		player.Name = encryptedName
+
+		_, err = db.Update(player)
+		if err != nil {
+			return err
+		}
+
+		err = db.Insert(&EncryptedPlayer{PlayerID: player.ID})
```

Comment:
Shouldn't you do this inside a transaction to prevent inconsistencies in the database?

Reply:
I think that is not necessary, once that if the EncryptedPlayers wasn't created the script already prevent this case

Reply:
Actually, I think it's better to be in another trx, since updating the encrypt table should not impact updating the player him/herself.

Reply:
So we would keep them in separate transactions? And if an error happens while inserting EncryptedPlayer, we would still save the player?

Reply:
@mathnogueira exactly